### ISSUE TITLE
Look into using `$and` instead of merging queries

### ISSUE THREAD

**Issue by mrfrase3** [fratzinger/feathers-casl#24]
Something that I do in one of my advanced feathers projects, when adding "security" filters, I add them to an `$and`, so that they won't conflict with the main query.

Say I have the following:

```js
// incoming query
{
  id: { $in: [1, 2, 3] },
  status: 'complete',
}
// casl abilities query:
{
  id: { $in: [2, 3, 4] },
  status: { $in: ['complete', 'pending', 'draft'] },
}
```
I'm not sure how smart the merging logic is, but what you can just do is:
```js
{
  id: { $in: [1, 2, 3] },
  status: 'complete',
  $and: [{
    id: { $in: [2, 3, 4] },
    status: { $in: ['complete', 'pending', 'draft'] },
  }],
}
```
All you need to do is check if `$and` exists as an array, otherwise initialise it, then push the ability query to the end of it.

I'm not certain it `$and` is supported by all adaptors, maybe having this as a togglable option instead of merging if you want to do more advanced filtering and your adaptor supports it?

**Comment by fratzinger** [comment]
Good call! There are a few edge cases that do not work with `mergeQuery` from `feathers-utils` as of now.
I will make the `$and` by default in the near future. It's supported by all main adapters (`feathers-memory`, `nedb`, `mongodb`, `mongoose`, `knex`, `objection`, `sequelize`).

Which adapter(s) do you use? :)

**Comment by mrfrase3** [comment]
mongoose, I'm a bit of a mongo fanboy at this point.

**Comment by fratzinger** [comment]
Please run `npm i feathers-casl@0.4.1-6` and check out, if it's working like you think it should. This will be integrated by PR #20.
I'll check that out by myself next week and if all goes well, I will merge it and make it `v0.5.0` then.

Any Feedback is highly appreciated! :)

**Comment by fratzinger** [comment]
released as [v0.5.0](https://github.com/fratzinger/feathers-casl/releases/tag/v0.5.0)
